src/cavendish_particle_tracks/_widget.py

```diff
@@ -41,66 +46,98 @@
+        self.btn_load.width = 200
+        self.cmb_add_particle = QComboBox()
+        self.cmb_add_particle.addItems(EXPECTED_PARTICLES)
+        self.cmb_add_particle.setCurrentIndex(0)
+        self.cmb_add_particle.currentIndexChanged.connect(
+            self._on_click_new_particle
+        )
+        self.btn_delete_particle = QPushButton("Delete particle")
+        self.btn_radius = QPushButton("Calculate radius")
+        self.btn_length = QPushButton("Calculate length")
+        self.btn_decayangle = QPushButton("Calculate decay angles")
+        self.btn_stereoshift = QPushButton("Stereoshift")
+        self.btn_save = QPushButton("Save")
+        self.btn_magnification = QPushButton("Magnification")
         # setup particle table
         self.table = self._set_up_table()
         self._set_table_visible_vars(False)
         self.table.selectionModel().selectionChanged.connect(
             self._on_row_selection_changed
         )
-
         # Apply magnification disabled until the magnification parameters are computed
         self.cal = QRadioButton("Apply magnification")
         self.cal.setEnabled(False)
 
         # connect callbacks
-        self.load.clicked.connect(self._on_click_load_data)
-        delete_particle.clicked.connect(self._on_click_delete_particle)
-        self.rad.clicked.connect(self._on_click_radius)
-        self.lgth.clicked.connect(self._on_click_length)
-        self.ang.clicked.connect(self._on_click_decay_angles)
-        self.stsh.clicked.connect(self._on_click_stereoshift)
+        self.btn_load.clicked.connect(self._on_click_load_data)
+        self.btn_delete_particle.clicked.connect(
+            self._on_click_delete_particle
+        )
+        self.btn_radius.clicked.connect(self._on_click_radius)
+        self.btn_length.clicked.connect(self._on_click_length)
+        self.btn_decayangle.clicked.connect(self._on_click_decay_angles)
+        self.btn_stereoshift.clicked.connect(self._on_click_stereoshift)
         self.cal.toggled.connect(self._on_click_apply_magnification)
-        save.clicked.connect(self._on_click_save)
-
-        self.mag.clicked.connect(self._on_click_magnification)
+        self.btn_save.clicked.connect(self._on_click_save)
+        self.btn_magnification.clicked.connect(self._on_click_magnification)
         # TODO: find which of thsese works
         # https://napari.org/stable/gallery/custom_mouse_functions.html
         # self.viewer.mouse_press.callbacks.connect(self._on_mouse_press)
         # self.viewer.events.mouse_press(self._on_mouse_click)
 
         # layout
-        self.setLayout(QVBoxLayout())
-        self.layout().addWidget(self.load)
-        self.layout().addWidget(self.cb)
-        self.layout().addWidget(delete_particle)
-        self.layout().addWidget(self.rad)
-        self.layout().addWidget(self.lgth)
-        self.layout().addWidget(self.ang)
+        self.viewer.window._qt_viewer.layerButtons.hide()  # This will break in napari 0.6.0
```

Comment:
```suggestion
        # self.viewer.window._qt_viewer.layerButtons.hide()  # This will break in napari 0.6.0
```

Reply:
Suggest to comment this out while we still allow for several `Stereoshift` and `Decay angles` layers to be created by repeatedly hitting the corresponding button.

Reply:
I've instead fixed this by not allowing the button by not allowing the user to open several stereoshift, mag and decay angle dialogs. It's not a long fix and might as well go for the long-term option if it's only 9 ish lines.

The user can still duplicate the layers of the stereoshift dialog, but this is fixed in the stereoshift PR and so is not corrected here.